inst/test_backends/test_MsBackend/test_spectra_variables.R

```diff
@@ -0,0 +1,286 @@
+})
+
+test_that("centroided", {
+    res <- centroided(be)
+    expect_true(length(res) == length(be))
+    expect_true(is.logical(res))
+    if (!isReadOnly(be)) {
+        tmp <- be
+        centroided(tmp) <- rep(FALSE, length(be))
+        expect_true(all(centroided(tmp) == FALSE))
+        centroided(tmp) <- rep(TRUE, length(be))
+        expect_true(all(centroided(tmp)))
+    }
+})
+
+test_that("collisionEnergy", {
+    res <- collisionEnergy(be)
+    expect_true(length(res) == length(be))
+    expect_true(is.numeric(res))
+    if (!isReadOnly(be)) {
+        tmp <- be
+        vals <- abs(rnorm(length(be)))
+        collisionEnergy(tmp) <- vals
+        expect_equal(unname(collisionEnergy(tmp)), unname(vals))
+    }
+})
+
+test_that("dataOrigin", {
+    res <- dataOrigin(be)
+    expect_true(length(res) == length(be))
+    expect_true(is.character(res))
+})
+
+test_that("dataStorage", {
+    res <- dataStorage(be)
+    expect_true(length(res) == length(be))
+    expect_true(is.character(res))
+})
+
+test_that("intensity", {
+    res <- intensity(be)
+    expect_true(is.list(res) || is(res, "NumericList"))
+    expect_true(length(res) == length(be))
+    if (!isReadOnly(be)) {
+        tmp <- be
+        vals <- lapply(lengths(res), function(z) abs(rnorm(z)))
+        intensity(tmp) <- vals
+        res <- intensity(tmp)
+        expect_equal(as.list(res), vals)
+    }
+})
+
+test_that("isCentroided", {
+    res <- isCentroided(be)
+    expect_true(length(res) == length(be))
+    expect_true(is.logical(res))
+})
+
+test_that("isEmpty", {
+    res <- isEmpty(be)
+    expect_true(length(res) == length(be))
+    expect_true(is.logical(res))
+    expect_equal(lengths(res) == 0, res)
+})
+
+test_that("isolationWindowLowerMz", {
+    res <- isolationWindowLowerMz(be)
+    expect_true(is.numeric(res))
+    expect_true(length(res) == length(res))
+    if (!isReadOnly(be)) {
+        tmp <- be
+        vals <- abs(rnorm(length(be)))
+        isolationWindowLowerMz(tmp) <- vals
+        expect_equal(unname(isolationWindowLowerMz(tmp)), unname(vals))
+    }
+})
+
+test_that("isolationWindowTargetMz", {
+    res <- isolationWindowTargetMz(be)
+    expect_true(is.numeric(res))
+    expect_true(length(res) == length(res))
+    if (!isReadOnly(be)) {
+        tmp <- be
+        vals <- abs(rnorm(length(be)))
+        isolationWindowTargetMz(tmp) <- vals
+        expect_equal(unname(isolationWindowTargetMz(tmp)), unname(vals))
+    }
+})
+
+test_that("isolationWindowUpperMz", {
+    res <- isolationWindowUpperMz(be)
+    expect_true(is.numeric(res))
+    expect_true(length(res) == length(res))
+    if (!isReadOnly(be)) {
+        tmp <- be
+        vals <- abs(rnorm(length(be)))
+        isolationWindowUpperMz(tmp) <- vals
+        expect_equal(unname(isolationWindowUpperMz(tmp)), unname(vals))
+    }
+})
+
+test_that("msLevel", {
+    res <- msLevel(be)
+    expect_true(is.integer(res))
+    expect_true(length(res) == length(res))
```

Comment:
Testing `length(res) == length(res)` should never fail :smile: 
```suggestion
    expect_identical(length(res), length(be))
```